src/remote_gym/repo_manager.py

```diff
@@ -0,0 +1,54 @@
+import hashlib
+from pathlib import Path
+from typing import Optional
+
+import git
+from fasteners import InterProcessLock
+
+
+def base64hash(string: str):
+    return hashlib.sha256(string.encode("utf-8")).hexdigest()
+
+
+def clone_and_checkout(directory: Path, repository: str, ref: Optional[str]):
+    # If the directory already exists, reuse the existing repo
+    if directory.exists():
+        repo = git.Repo(directory)
+        repo.remotes.origin.fetch(tags=True)
+    else:
+        # Clone the repository if it doesn't exist
+        repo = git.Repo.clone_from(repository, directory)
+
+    # Use the masters remote head by default
+    if ref is None:
+        ref = "HEAD"
+
+    if ("origin/" + ref) in repo.references:
```

Comment:
Is this the place where it differentiates between tags and commit hashes?

If yes, then it might make sense to add to the documentation of RepoManager.get, that branch names are also supported. Currently the documentation and parameter names do not indicate that passing a branch is possible.

Reply:
Also, it seems (based on the tests) that you support branch names and commit hashes. I have not seen where tag support is implemented.

Reply:
Branches, tags and commit ids are all valid references, no explicit support is required here afaik. Branches however, unlike tags for some reasons, can be local. We don't want that, thus the "origin/" test to make sure it always uses the one from the origin.